lib/OpenQA/WebAPI/Controller/API/V1/Search.pm

```diff
@@ -149,10 +149,23 @@ sub _search_job_templates {
     my @results;
     my $last_group = -1;
     my $like       = {like => "%${keywords}%"};
-    my $templates  = $self->schema->resultset('JobTemplates')->search({-or => {name => $like, description => $like}})
-      ->slice(0, $limit);
+
+    # Take into account names of test suites in cases where the job template itself has no name or description,
+    # but is based off of a test suite
+    my $templates = $self->schema->resultset('JobTemplates')->search(
+        {-or => ['me.name' => $like, 'me.description' => $like, 'test_suite.name' => $like]},
+        {
+            join     => 'test_suite',
+            order_by => {-desc => 'group_id'}})->slice(0, $limit);
     while (my $template = $templates->next) {
-        my $contents = $template->name . "\n" . $template->description;
+        my $contents;
+        my $description = $template->description;
+        if (my $template_name = $template->name) {
+            $contents = $template_name . "\n" . $description;
+        }
+        else {
+            $contents = $template->test_suite->name . "\n" . $description;
```

Comment:
I think you should be able to combine the two lines, e.g. use `($template->name // $template->test_suite->name)`.

However I think this would be even better defined as a method for the template class itself to give back this fallback. Would it be problematic to just let the "name" method return the testsuite name as fallback?

Reply:
It would make the code less obvious, also considering we're still searching test suites in the same context. The same goes for the existing job settings code, it would add confusion.

Reply:
I tried this  `my $template_name = $template->name // $template->test_suite->name;`, but it always return the result of `$template->name`, so I change it to `? :` this kind.

IMHO, the `name` method returns the test suite name as fallback may cause the confuse.